### PR TITLE
Add convert func and fix dependency in autoware_iv_auto_msgs_converter

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

add convert func, and also fix dependency of package.xml

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Code is properly formatted
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code is properly formatted
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by kenji-miyake** [comment]
Could you make the PR title a bit more descriptive and fix the typo?

File: autoware_iv_auto_msgs_converter/include/autoware_iv_auto_msgs_converter/autoware_iv_auto_msgs_converter.hpp [hunk 1]
```diff
@@ -75,6 +75,26 @@ inline auto convert(const autoware_auto_system_msgs::msg::AutowareState & state)
...
+  } else if (state.state == autoware_system_msgs::msg::AutowareState::PLANNING) {
+    auto_state.state = autoware_auto_system_msgs::msg::AutowareState::PLANNING;
+  } else if (state.state == autoware_system_msgs::msg::AutowareState::WAITING_FOR_ENGAGE) {
+    auto_state.state = autoware_auto_system_msgs::msg::AutowareState::WAITING_FOR_ENGAGE;
+  } else if (state.state == autoware_system_msgs::msg::AutowareState::DRIVING) {
+    auto_state.state = autoware_auto_system_msgs::msg::AutowareState::DRIVING;
+  } else if (state.state == autoware_system_msgs::msg::AutowareState::ARRIVAL_GOAL) {
+    auto_state.state = autoware_auto_system_msgs::msg::AutowareState::ARRIVED_GOAL;
+  }
+  auto_state.state = autoware_auto_system_msgs::msg::AutowareState::FINALIZING;
```
Comment (isamu-takagi) [inline autoware_iv_auto_msgs_converter/include/autoware_iv_auto_msgs_converter/autoware_iv_auto_msgs_converter.hpp:100]: It looks like this line always overwrite the state.

Comment (takayuki5168) [reply autoware_iv_auto_msgs_converter/include/autoware_iv_auto_msgs_converter/autoware_iv_auto_msgs_converter.hpp:100]: :innocent: 
https://github.com/tier4/AutowareArchitectureProposal_msgs/pull/12/commits/114f74dc9f267b60dd9c5face73da8e226806e88

Actually I don't use this function, but someone may need this function so let's merge this PR when it's ready.

**Comment by takayuki5168** [comment]
No need to merge this PR, so I close

**Pull request closed without merging**